dlt/destinations/impl/lancedb/lancedb_client.py

```diff
@@ -0,0 +1,698 @@
+                    else:
+                        embedding_fields = None
+                        vector_field_name = None
+                        id_field_name = None
+                        embedding_model_func = None
+                        embedding_model_dimensions = None
+
+                    table_schema: TArrowSchema = make_arrow_table_schema(
+                        table_name,
+                        schema=self.schema,
+                        type_mapper=self.type_mapper,
+                        embedding_fields=embedding_fields,
+                        embedding_model_func=embedding_model_func,
+                        embedding_model_dimensions=embedding_model_dimensions,
+                        vector_field_name=vector_field_name,
+                        id_field_name=id_field_name,
+                    )
+                    fq_table_name = self.make_qualified_table_name(table_name)
+                    self.create_table(fq_table_name, table_schema)
+
+        self.update_schema_in_storage()
+
+    @lancedb_error
+    def update_schema_in_storage(self) -> None:
+        records = [
+            {
+                "version": self.schema.version,
+                "engine_version": self.schema.ENGINE_VERSION,
+                "inserted_at": str(pendulum.now()),
+                "schema_name": self.schema.name,
+                "version_hash": self.schema.stored_version_hash,
+                "schema": json.dumps(self.schema.to_dict()),
+            }
+        ]
+        fq_version_table_name = self.make_qualified_table_name(self.schema.version_table_name)
+        write_disposition = self.schema.get_table(self.schema.version_table_name).get(
+            "write_disposition"
+        )
+        upload_batch(
+            records,
+            db_client=self.db_client,
+            table_name=fq_version_table_name,
+            write_disposition=write_disposition,
+        )
+
+    @lancedb_error
+    def get_stored_state(self, pipeline_name: str) -> Optional[StateInfo]:
+        """Loads compressed state from destination storage by finding a load ID that was completed."""
+        fq_state_table_name = self.make_qualified_table_name(self.schema.state_table_name)
+        fq_loads_table_name = self.make_qualified_table_name(self.schema.loads_table_name)
+
+        state_records = (
+            self.db_client.open_table(fq_state_table_name)
+            .search()
+            .where(f'pipeline_name = "{pipeline_name}" ORDER BY _dlt_load_id DESC')
+            .to_list()
+        )
+        if len(state_records) == 0:
+            return None
+        for state in state_records:
+            load_id = state["_dlt_load_id"]
+            # If there is a load for this state which was successful, return the state.
+            if (
+                self.db_client.open_table(fq_loads_table_name)
+                .search()
+                .where(f'load_id = "{load_id}"')
```

Comment:
you need to filter by status = 0 also, other entries are not considered successful loads

Reply:
You can check out get_stored_state from the job_client_impl. You need to somehow replicate the join that is done there with the tables that you have here. The code of the other vector DBs will help here.